aws/eks-anywhere
# Make DiskOffering in CloudStackMachineConfig optional

*Issue #, if available:*

*Description of changes:*
`DiskOffering: {}` is present when marshaling a CloudStackMachineConfig without defining it. This might break the backward compatibility and cause unwanted rolling upgrade. 

*Testing (if applicable):*
Unit testing
Manual testing
1. Create a cluster without DiskOffering
2. Upgrade the cluster with DiskOffering
3. Upgrade the cluster without DiskOffering 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
